src/main/java/net/neoforged/art/internal/ClassProviderImpl.java

```diff
@@ -179,9 +179,25 @@ public Collection<? extends IFieldInfo> getFields() {
             return fieldsView;
         }
 
+        @Deprecated
         @Override
         public Optional<? extends IFieldInfo> getField(String name) {
-            return fields == null ? Optional.empty() : Optional.ofNullable(this.fields.get(name));
+            if (this.fields != null) {
```

Comment:
Given that everything currently uses this method, I don't like the performance implications of it. You should instead create a new map that stores raw name -> info.